### PR TITLE
New task on error

### PR DESCRIPTION
**Pull request by jpa198**
When a task errors a new task is made instead of increasing the retry count

### CODE REVIEW

File: scripts/cybershake/auto_submit.py [hunk 1]
```diff
@@ -152,16 +153,6 @@ def update_tasks(
                 error="Disappeared from squeue. Reset to created.",
```
Comment (jasonmotha) [inline scripts/cybershake/auto_submit.py]: update error message

File: scripts/management/update_mgmt_db.py [hunk 2]
```diff
@@ -1,119 +0,0 @@
...
-A script that updates a slurm mgmt db and updates the status of a task
-"""
-
-import argparse
-import qcore.constants
-from scripts.management import db_helper
-import qcore.constants as const
-
-
-def update_db(db, process, status, job=None, run_name=None, error=None):
```
Comment (jasonmotha) [inline scripts/management/update_mgmt_db.py]: I think we need to restore a minimal version of this file

File: scripts/management/MgmtDB.py [hunk 3]
```diff
@@ -160,22 +165,22 @@ def _check_dependancy_met(self, task, logger=workflow_logger.get_basic_logger())
...
-            (entry.status, entry.job_id, entry.retries),
+            (self.col_status, self.col_job_id),
+            (entry.status, entry.job_id),
         ):
             if value is not None:
                 cur.execute(
                     "UPDATE state SET {} = ?, last_modified = strftime('%s','now') "
-                    "WHERE run_name = ? AND proc_type = ?".format(field),
-                    (value, entry.run_name, entry.proc_type),
+                    "WHERE run_name = ? AND proc_type = ? and status != ?".format(field),
```
Comment (jasonmotha) [inline scripts/management/MgmtDB.py]: Change to less than. So statuses can only progress forward

File: e2e_tests/E2ETests.py [hunk 4]
```diff
@@ -440,6 +441,7 @@ def cancel_running(self, proc_types: List[const.ProcessType]):
                     "Checkpoint testing: Cancelling job-id {} "
                     "for {} and process type {}".format(entry[1], entry[2], entry[0])
                 )
+                self.canceled_running.append(str(entry[1]))
```
Comment (claudio525) [inline e2e_tests/E2ETests.py]: Why as str and not int?

File: scripts/management/update_mgmt_db.py (hunk 2 above)
Comment (claudio525) [reply scripts/management/update_mgmt_db.py]: Do we use this anywhere in our workflow code? Or only for manual updates?

File: e2e_tests/E2ETests.py (hunk 4 above)
Comment (jpa198) [reply e2e_tests/E2ETests.py]: They get put in an sql query later on, and I think you have to use strings for that

File: scripts/management/update_mgmt_db.py (hunk 2 above)
Comment (jpa198) [reply scripts/management/update_mgmt_db.py]: This is only used for manual updates. The alternative method is to run add_to_mgmt_queue while queue_monitor is running. 
This should also have a diff for the file I re added.

File: e2e_tests/E2ETests.py (hunk 4 above)
Comment (jasonmotha) [reply e2e_tests/E2ETests.py]: Prepared queries (using ?) convert to str for you.

**Comment by jpa198** [comment]
I ran an end to end test with a raised cancel rate and retry limit and everything worked.

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,13 @@ Slurm Ground Motion Workflow
 # Changelog
 (Based on https://wiki.canterbury.ac.nz/download/attachments/58458136/CodeVersioning_v18p2.pdf?version=1&modificationDate=1519269238437&api=v2 )
 
+## [19.6.1] - 2019-06-05 -- Error task changes
+### Added
+    - A wrapper for auto_submit and queue_monitor is now available 
+    - LF and HF may now be converted to BB without the over, and IM calculations subsequently performed on them
```
Comment (jasonmotha) [inline CHANGELOG.md]: other

Comment (jpa198) [reply CHANGELOG.md]: England much hard

**Pull request merged**